src/dirent.rs

```diff
@@ -0,0 +1,58 @@
+use {Result, Error, Errno, NixPath};
+use errno;
+use libc::{self, DIR, c_long};
+
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+use std::os::unix::io::RawFd;
+
+pub struct Dir {
+    handle: *mut DIR,
+}
+
+impl Drop for Dir {
+    fn drop(&mut self) {
+        unsafe { libc::closedir(self.handle) };
+    }
+}
+
+pub fn opendir<P: ?Sized + NixPath>(name: &P) -> Result<Dir> {
+    let dirp = try!(name.with_nix_path(|cstr| unsafe { libc::opendir(cstr.as_ptr()) }));
+    if dirp.is_null() {
+        Err(Error::last().into())
+    } else {
+        Ok(Dir { handle: dirp })
+    }
+}
+
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+pub fn fdopendir(fd: RawFd) -> Result<Dir> {
+    let dirp = unsafe { libc::fdopendir(fd) };
+    if dirp.is_null() {
+        Err(Error::last().into())
+    } else {
+        Ok(Dir { handle: dirp })
+    }
+}
+
+pub fn readdir<'a>(dir: &'a mut Dir) -> Result<Option<&'a libc::dirent>> {
```

Comment:
To obtain `d_name` I use:

```
let name = OsStr::from_bytes(unsafe {
    &mem::transmute::<[i8; 256], [u8; 256]>(entry.d_name)
});
```

any idea how I could how add a convenience wrapper to this library?